.github/workflows/refresh_deployment.yaml

```diff
@@ -0,0 +1,64 @@
+name: Refresh EC2 FINOS Waltz deployment
+
+on:
+  workflow_dispatch:
+  #schedule:
+    # Runs everyday at 00:00. (see https://crontab.guru)
+    #- cron: '0 0 * * *'
+
+jobs:
+  build:
+    name: Trigger Juju refresh on EC2 Deployment
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure SSH
+        run: |
+          mkdir -p ~/.ssh/
+          echo "$SSH_KEY" > ~/.ssh/deployment.key
+          chmod 600 ~/.ssh/deployment.key
+          cat >>~/.ssh/config <<END
+          Host deployment
+            HostName $SSH_HOST
+            User $SSH_USER
+            IdentityFile ~/.ssh/deployment.key
+            StrictHostKeyChecking no
+          END
+        env:
+          SSH_USER: ${{ secrets.SSH_USER }}
+          SSH_KEY: ${{ secrets.SSH_KEY }}
+          SSH_HOST: ${{ secrets.SSH_HOST }}
+
+      - name: Refresh Deployment
+        run: |
+          ssh deployment '
+            set -x
+
+            # Running juju status will show us the current revisions of the charms.
+            juju status --relations
+
+            juju refresh finos-waltz --channel=edge
+            juju refresh waltz-ingress --channel stable
+
+            # Running juju status will show us the current revisions of the charms.
+            juju status --relations
+          '
+      - name: Check Waltz Public Instance status
+        run: |
+          wait_for_curl() {
+            url="$1"
+            for i in {1..30}; do
+              status=$(curl -s -o /dev/null -w "%{http_code}" "$url")
+              if [ "$status" = "200" ]; then
+                break
+              fi
+              if [ "$i" -eq 30 ]; then
+                echo "Failed to get 200 response status from $url. Got status $status."
+                exit 1
+              fi
+              sleep 10
+            done
+          }
+
+          wait_for_curl "http://demo.waltz.finos.org/"
+
+          echo "Waltz is reachable"
```

Comment:
you should add an empty newline here, so we don't have that "No newline at end of file" error.

Reply:
Empty line added